### ISSUE TITLE
[Bug] User in a non-root directory breaks avatar creator

### ISSUE THREAD

**Issue by shiftyscales** [Frooxius/NeosPublic#385]
Steps to replicate:
- Ensure a player is in a non-root slot, e.g. parent them to a new slot, or use an avatar spawn point in a non-root place
- Import a rigged avatar
- Open an avatar creator, and attempt to create an avatar

From here- the outcomes split depending on one common factor- the user's current avatar.

If the user is using an avatar without a full rig- a "head and hands" avatar, the avatar creator will simply not respond to user input:
https://youtu.be/xn5azGoAmCg

If the user is using any avatar with a rigged body, attempting to use the alignment buttons will position them strangely- including occasionally, on the user's own hands in the case of the 'try align hands'.

When the user attempts to create the avatar, even if entirely positioned over it properly- the user's avatar will be deleted, and the avatar will not be created, leaving the model, and the avatar creator in the world:
https://youtu.be/Crsv0vEMOFw

**Comment by Enverex** [comment]
Tagging myself as I was the one that unfortunately discovered this. You have to use the emergency respawn method to recover as you can no-longer access anything when it happens.

**Comment by shiftyscales** [comment]
I'd previously also observed this phenomenon with PurpleJuice in their home world a couple of months back, but was unable to replicate it/figure it out at the time. Thanks again for the feedback that helped me track it down, @Enverex.

**Comment by Frooxius** [comment]
This is interesting finding thanks! Does it actually matter if the user is in non-root slot or does it happen even if the avatar creator is in non-root? When you spawn it out, is it under root or under the same slot with you?

What about the object that you're creating the avatar from, is that under root or under the non-root slot as well?

**Comment by shiftyscales** [comment]
Avatar creator always spawns in the root regardless of the player's place in the hierarchy, models are imported into the same hierarchy as the user.

I can test different iterations of moving the avatar creator, and model to import, but they aren't situations that should come up in normal use.

I will update shortly with my findings.

**Comment by Frooxius** [comment]
I tried to replicate this, but I can't seem to get it to happen on my end, the avatar creator works normally even if I'm parented under a non-root slot. There must be something more going on.

**Comment by shiftyscales** [comment]
There is, actually. I just discovered it is tied to using the 'align' options. Hitting create while manually aligning works as intended, attempting to auto-align is what causes it to break in that situation.

Following up on the other scenarios:
If the user, and model are in the root, but the avatar creator is reparented into non-root slot, it works as intended.

If the avatar creator is moved into the same non-root slot as the user, and model after it is created, it will break in the same way as initially described.

If the user is in a non-root slot, and the model is on the root while the avatar creator is spawned in, it works as intended.

**Comment by Enverex** [comment]
I could record a video of this happening (from joining the session to being deleted) so you can see the step by step process and perhaps spot what may be the cause, if that would help? You may spot something somewhere along the line that is being missed when just describing the problem.

**Comment by shiftyscales** [comment]
I've recorded and sent numerous samples of the bug in action to Froox, already. That's always a good idea to record video where possible though.

**Comment by Enverex** [comment]
I was tempted to record one myself because I noticed your comment that "If the user is in a non-root slot, and the model is on the root while the avatar creator is spawned in, it works as intended." which I think is actually the situation where it actually breaks for me, but I need to double check.

**Comment by Frooxius** [comment]
I see thanks for the info! From the sound of it, it seems that it doesn't matter where the user is parented, but rather where the model that's being made into an avatar is, I think that's why I had trouble replicating it, because the model was always under root. 

Can you test just this scenario? Everything under the root, except the avatar which is not under root. Will it break in that case?

**Comment by shiftyscales** [comment]
From a new basic empty session I:
- Imported the Y_Bot.fbx
- Created an empty object with the DevToolTip
- Parented the Y_Bot.fbx under it by dragging and dropping it from the root
- Made an avatar creator
- Placed the head and hands accordingly
- Hit create

Outcome:
The avatar was created successfully. Nothing broke.

I then repeated this test, but with myself parented under a non-root slot at the time of model import, then moving myself to the root after import, and prior to the creation of the avatar creator. It did not break.

Repeating again with the avatar imported, and avatar creator spawned before moving myself into the root also didn't break.

I then repeated the test with myself on a second empty object separate from the same hierarchy as the fbx. Nothing broke.

I then once again ran the usual test of creating an empty object, parenting myself under it, importing the avatar, then creating the avatar creator leading to the usual, and reported breakage.

So it appears to definitively be in the case where a user is in a non-root slot with the avatar they've imported in the same hierarchy as themself.

A user in a non-root slot does not cause the creator to break as long as the model being turned into an avatar is on a separate hierarchy from the user.
